invenio_files_rest/tasks.py

```diff
@@ -50,16 +55,109 @@ def progress_updater(size, total):
+        celery.conf['CELERYBEAT_SCHEDULE'] entry of this task. The argument is
+        a dictionary that will be passed as arguments to the
+        `datetime.timedelta` class.
+    :param int max_count: Max count of files of a single batch. When set to `0`
+        it's automatically calculated to be distributed equally through the
+        number of total batches.
+    :param int max_size: Max size of a single batch in bytes. When set to `0`
+        it's automatically calculated to be distributed equally through the
+        number of total batches.
+    """
+    assert max_count is not None or max_size is not None
+    frequency = timedelta(**frequency) if frequency else timedelta(days=30)
+
+    if batch_interval:
+        batch_interval = timedelta(**batch_interval)
+    else:
+        celery_schedule = current_celery.conf.get('CELERYBEAT_SCHEDULE', {})
+        batch_interval = batch_interval or next(
+            (v['schedule'] for v in celery_schedule.values()
+             if v.get('task') == schedule_checksum_verification.name), None)
+    if not batch_interval or not isinstance(batch_interval, timedelta):
+        raise Exception(u'No "batch_interval" could be decided')
+
+    total_batches = frequency.total_seconds() / batch_interval.total_seconds()
+
+    files = load_or_import_from_config(
+        'FILES_REST_CHECKSUM_VERIFICATION_FILES_QUERY')()
+    files = files.order_by(FileInstance.last_check_at)
+
+    if max_count is not None:
+        all_files_count = files.count()
+        min_count = int(math.ceil(all_files_count / total_batches))
+        max_count = min_count if max_count == 0 else max_count
+        if max_count < min_count:
+            warn(u'The "max_count" you specified ({0}) is smaller than the'
+                 'minimum batch file count required ({1}) in order to achieve '
+                 'the file checks over the specified period ({2}).'
+                 .format(max_count, min_count, frequency))
+        files = files.limit(max_count)
+
+    if max_size is not None:
+        all_files_size = db.session.query(
+            sa.func.sum(FileInstance.size)).scalar()
+        min_size = int(math.ceil(all_files_size / total_batches))
+        max_size = min_size if max_size == 0 else max_size
+        if max_size < min_size:
+            warn(u'The "max_size" you specified ({0}) is smaller than the'
+                 'minimum batch total file size required ({1}) in order to '
+                 'achieve the file checks over the specified period ({2}).'
+                 .format(max_size, min_size, frequency))
+
+    files = files.yield_per(1000)
+    scheduled_file_ids = []
+    total_size = 0
+    for f in files:
+        # Add at least the first file, since it might be larger than "max_size"
+        scheduled_file_ids.append(str(f.id))
+        total_size += f.size
+        if max_size and max_size < total_size:
+            break
+    print('Sending this files for checksum:', scheduled_file_ids)
```

Comment:
Is the print supposed to be here? If yes, then perhaps better to use the task logger.